listenbrainz/webserver/static/js/src/RecentListens.tsx

```diff
@@ -263,6 +263,23 @@ export default class RecentListens extends React.Component<
           return;
         }
       }
+      // The websocket message received contains the recording_msid as a top level key.
+      // Therefore, we need to shift it json.track_metadata.additional_info.
+      if (!("recording_msid" in json.track_metadata.additional_info)) {
```

Comment:
This can throw error is a variety of cases if the structure isn't exactly `json.track_metadata.additional_info`:
- cannot access property of undefined if track_metadata isn't there
- cannot use 'in' operator if right-side is undefined


Instead, you can use a safe method like [lodash's `has`](https://lodash.com/docs/#has):
`if(!has(json, "track_metadata.additional_info.recording_msid")){…`

Reply:
done